x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/EsqlPartitioningIT.java

```diff
@@ -64,12 +68,18 @@ public static Iterable<Object[]> parameters() {
                     new Case("| STATS SUM(a)", "DOC"),
                     new Case("| MV_EXPAND a | STATS SUM(a)", "DOC"),
                     new Case("| WHERE a == 1 | STATS SUM(a)", "SEGMENT"),
-                    new Case("| WHERE a == 1 | MV_EXPAND a | STATS SUM(a)", "SEGMENT"), }) {
+                    new Case("| WHERE a == 1 | MV_EXPAND a | STATS SUM(a)", "SEGMENT"),
+                    new Case("| WHERE MATCH(a, \"1\")", "SHARD"),
+                    new Case("| WHERE a:\"1\"", "SHARD"),
+                    new Case("| WHERE MATCH(a, \"2\") | SORT _score DESC", "SHARD", true),
+                    new Case("| WHERE MATCH(a, \"3\") | SORT _score DESC | LIMIT 10", "SHARD", true),
+                    new Case("| WHERE MATCH(a, \"3\") OR MATCH(a, \"4\") | SORT _score DESC | LIMIT 10", "SHARD", true),
```

Comment:
Should using scoring be random? Same for the full text function being used (QSTR, KQL)

Reply:
I thought it simplest to just have some straightforward scoring and some non-scoring scenarios. The non-scoring cannot use "SORT _score", for obvious reasons, and when I tried to make the sort conditional it just obfuscated the scenario. I added some scenarios for KQL and QSTR. This should be sufficient.

Reply:
Sounds good to me!